crates/matrix-sdk-crypto/src/identities/room_identity_state.rs

```diff
@@ -219,16 +227,11 @@ impl<R: RoomIdentityProvider> RoomIdentityState<R> {
 
             // Changed the type of bad - report so can change the message
             (PinViolation, VerificationViolation) |
-            (VerificationViolation, PinViolation) => Some(self.set_state(user_id, new_state)),
+            (VerificationViolation, PinViolation) |
 
-            // good -> good - don't report - no message needed in either case
+            // Changed the type of good - report so can change the message
             (Pinned, Verified) |
-            (Verified, Pinned) => {
-                // The state has changed, so we update it
-                self.set_state(user_id, new_state);
-                // but there is no need to report a change to the UI
-                None
-            }
+            (Verified, Pinned) => Some(self.set_state(user_id, new_state)),
```

Comment:
This whole match has now become `if old_state == new_state` (right?) so let's just change it to that. All those comments were to explain something irritatingly complicated that you have now removed :-)

Reply:
Ha, I was so busy tweaking it I didn't realise I can simplify the whole thing, love it!